application/src/main/java/org/togetherjava/tjbot/commands/moderation/report/ReportCommand.java

```diff
@@ -0,0 +1,175 @@
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        reportedMessage = event.getTarget().getContentRaw();
+        reportedMessageUrl = event.getTarget().getJumpUrl();
+
+        TextInput body = TextInput.create(MESSAGE_INPUT, "Message", TextInputStyle.PARAGRAPH)
+            .setPlaceholder("Reason for reporting")
+            .setRequiredRange(10, 200)
+            .build();
+
+        Modal modal = Modal.create(generateComponentId(), "Report").addActionRow(body).build();
+
+        event.replyModal(modal).queue();
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        long userID = event.getUser().getIdLong();
+
+        if (handleIsOnCooldown(userID, event)) {
+            return;
+        }
+        authorToLastReportInvocation.put(userID, Instant.now());
+        event.deferReply().setEphemeral(true).queue();
+
+        long guildID = Objects.requireNonNull(event.getGuild(), "Could not retrieve the guildId.")
+            .getIdLong();
+        Optional<TextChannel> modMailAuditLog = getModMailChannel(event.getJDA(), guildID);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    guildID, configModMailChannelPattern);
+            return;
+        }
+        String modalMessage = event.getValue(MESSAGE_INPUT).getAsString();
+        MessageCreateAction message =
+                createModMessage(modalMessage, userID, modMailAuditLog.orElseThrow());
+
+        sendMessage(event, message);
+
+        event.getHook().editOriginal("Thank you for your feedback!").queue();
+
+    }
+
+    private boolean handleIsOnCooldown(long userID, ModalInteractionEvent event) {
+        if (!isChannelOnCooldown(userID)) {
+            return false;
+        }
+        event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
+    }
+
+    private boolean isChannelOnCooldown(long userId) {
+        return Optional.ofNullable(authorToLastReportInvocation.getIfPresent(userId))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    private Optional<TextChannel> getModMailChannel(JDA jda, long guildID) {
+        return jda.getGuildById(guildID)
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+    }
+
+    private MessageCreateAction createModMessage(String modalMessage, long userID,
```

Comment:
the name of `createModMessage` and `createReportMessage` suggests that they would be about sth entirely different, but they are not. not at all. this is confusing.

id suggest to get rid of one of the two methods and inline the other. for example moving `createReportMessage` up to here.

Reply:
Maybe renaming it better? Because `createReportMessage ` is just there so there isn't a large embed in that call so maybe

`createReportMessageEmbed` ?

Reply:
But you can also just, you know, create a variable and send that:

```java
Embed embed = ...
  ...
  ... // create large embed

channel.sendMessage(embed) // and now send it
```

Reply:
Done